src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Herbivore && target.hidden !== true) {
+      target.health -= 50;
+
+      if (target.health === 0) {
+        const index = Animal.alive.indexOf(target);
+
+        Animal.alive.splice(index, 1);
```

Comment:
Better filter the animals and leave only animals with positive health
![image](https://github.com/mate-academy/js_herbivores_and_carnivores/assets/96409563/0107b393-0802-48da-8402-73c8a8b26c8e)